types/react-router/index.d.ts

```diff
@@ -61,7 +61,7 @@ declare module 'react-router' {
 
   interface RouteProps {
     location?: H.Location;
-    component?: React.SFC<RouteComponentProps<any> | void | {}> | React.ComponentClass<RouteComponentProps<any> | void | {}>;
```

Comment:
I can change this to `React.SFC<RouteComponentProps<any> | {}>`.
But this may fail [PreventingTransitions test](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-router/test/PreventingTransitions.tsx#L25).

I removed `RouteComponentProps` because [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15862#discussion_r113478724).
(cc: @andy-ms Do you have any idea?)

I do not mind anything for this ( `<RouteComponentProps<any> | void | {}> ` or `React.SFC<void | {}> ` or `<RouteComponentProps<any> | {}> `) 
and I feel `<RouteComponentProps<any> | void | {}> ` is better.